From: Xianchao <xlong@rtr.ai>
Date: Wed, 5 Jan 2022 16:04:36 -0500
Subject: cherry pick fix on a numerical issue from melodic

---
 src/frames.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/frames.cpp b/src/frames.cpp
index 94f69de..e45e8fc 100644
--- a/src/frames.cpp
+++ b/src/frames.cpp
@@ -417,15 +417,13 @@ double Rotation::GetRotAngle(Vector& axis,double eps) const {
         return angle; // return 180 deg rotation
     }
 
-    // If the matrix is slightly non-orthogonal, `f` may be out of the (-1, +1) range.
-    // Therefore, clamp it between those values to avoid NaNs.
     double f = (data[0] + data[4] + data[8] - 1) / 2;
-    angle = acos(std::max(-1.0, std::min(1.0, f)));
 
     x = (data[7] - data[5]);
     y = (data[2] - data[6]);
     z = (data[3] - data[1]);
     axis = KDL::Vector(x, y, z);
+    angle = atan2(axis.Norm()/2,f);
     axis.Normalize();
     return angle;
 }
